### PR TITLE
Kyverno mutate eks-prow-build-cluster to pull images from ecr

### PR DESCRIPTION
**Pull request by junshun**
This uses Kyverno to address issue #4928 

This was tested using a fake pod manifest and applying the kyverno policy in clusterpolicy.yaml
```
apiVersion: v1
kind: Pod
metadata:
  name: nginx
spec:
  containers:
  - name: nginx
    image: gcr.io/k8s-prow/nginx:1.14.1
  - name: nginx-staging
    image: gcr.io/k8s-staging-test-infra/nginx:1.14.1
  - name: nginx-others
    image: gcr.io/k8s-others-test-infra/nginx:1.14.1
  - name: nginx-fail
    image: gcr.io/k8s-staging/nginx:1.14.1
``` 
It was able to mutate the images to 
```
k describe pod | grep Image:
    Image:          468814281478.dkr.ecr.us-east-2.amazonaws.com/k8s-prow/nginx:1.14.1
    Image:          468814281478.dkr.ecr.us-east-2.amazonaws.com/k8s-staging-test-infra/nginx:1.14.1
    Image:          gcr.io/k8s-others-test-infra/nginx:1.14.1
    Image:          gcr.io/k8s-staging/nginx:1.14.1
```

### CODE REVIEW

**Comment by upodroid** [comment]
/ok-to-test

Thanks. Please sign the CLA

**Comment by upodroid** [comment]
@xmudrii Hi, can you deploy these manifests (except the clusterpolicy.yaml file)?

/assign @xmudrii

File: infra/aws/terraform/prow-build-cluster/resources/kyverno/kyverno.yaml
Comment (xmudrii) [inline infra/aws/terraform/prow-build-cluster/resources/kyverno/kyverno.yaml:1]: Please change PodDisruptionBudget apiVersion to `policy/v1`. `policy/v1beta1` doesn't exist in Kubernetes 1.25:

```
resource mapping not found for name: "kyverno" namespace: "kyverno" from "kyverno.yaml": no matches for kind "PodDisruptionBudget" in version "policy/v1beta1"
```

**Comment by xmudrii** [comment]
@upodroid @junshun Applied:

```
NAME                                          READY   STATUS    RESTARTS        AGE
kyverno-6bfcb5cdfc-99dgp                      1/1     Running   0               2m35s
kyverno-6bfcb5cdfc-lkhgk                      1/1     Running   2 (2m25s ago)   2m35s
kyverno-6bfcb5cdfc-lldts                      1/1     Running   0               2m35s
kyverno-cleanup-controller-85cf97fb88-j9h7v   1/1     Running   0               2m35s
```

(pod restarts are due to missing CRDs; server-side apply or `kubectl create` is required because of the annotation length)

File: infra/aws/terraform/prow-build-cluster/resources/kyverno/kyverno.yaml
Comment (junshun) [reply infra/aws/terraform/prow-build-cluster/resources/kyverno/kyverno.yaml:1]: done. Added section to readme about this as well.

**Comment by xmudrii** [comment]
@junshun The PR needs a rebase (probably because of `namespaces.yaml`). Can you please rebase it and then we can merge it?

**Review by xmudrii — APPROVED** [review]

**Comment by xmudrii** [comment]
/assign @ameukam 
for approval

**Comment by dims** [comment]
/hold

Let's make sure we won't see this problem in our build cluster before we land this change - https://github.com/kyverno/kyverno/issues/5830

**Comment by xmudrii** [comment]
We should eventually get back to this
/remove-lifecycle stale

**Comment by sftim** [comment]
How's this going?

**Comment by xmudrii** [comment]
@sftim It got blocked on https://github.com/kubernetes/k8s.io/pull/4994#issuecomment-1487029080, but given it worked out without this for a while now, we need to reevaluate if this is something we really need. @upodroid Do you maybe have any feedback on this?

**Pull request closed without merging**
